### PR TITLE
Add test to trigger division by zero handler

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Add test to trigger division by zero handler.

Resolves: #668 

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

### CODE REVIEW

File: tests/api_test/api_test.cpp [hunk 1]
```diff
@@ -288,8 +288,24 @@ ring_buffer_api_test(ebpf_execution_type_t execution_type)
     });
 }
 
+void
+divide_by_zero_test(ebpf_execution_type_t execution_type)
```
Comment (dthaler) [inline tests/api_test/api_test.cpp:292]: This name and prototype look identical to divide_by_zero_test in end_to_end.cpp.   What is the difference?   At least add // comments to both places explaining what the difference is from the other one?

Comment (dthaler) [reply tests/api_test/api_test.cpp]: I see you renamed it to end in _km which is ok (and I assume that means yes the difference is just UM vs KM), but would still like better comments that cross reference the two places.  E.g., rename the other one to end in _um and add // comments above both of them such as:
```
api_test.cpp:
// See also divide_by_zero_test_um in end_to_end.cpp for the user-mode equivalent.
void
divide_by_zero_test_km(ebpf_execution_type_t execution_type)
...
end_to_end.cpp:
// See also divide_by_zero_test_km in api_test.cpp for the kernel-mode equivalent.
void
divide_by_zero_test_um(ebpf_execution_type_t execution_type)
```

File: libs/execution_context/ebpf_program.c [hunk 2]
```diff
@@ -671,6 +671,8 @@ _ebpf_program_load_byte_code(
     // failing.
     ubpf_toggle_bounds_check(program->code_or_vm.vm, false);
 
+    ubpf_set_error_print(program->code_or_vm.vm, ebpf_log_function);
```
Comment (shankarseal) [inline libs/execution_context/ebpf_program.c:674]: oh so was there a bug? Shouldnt the UM code invoke this function in EC as well?

Comment (Alan-Jowett) [reply libs/execution_context/ebpf_program.c:674]: Because of this:
https://github.com/microsoft/ebpf-for-windows/blob/980646a793c3a18e5175b3a7b012e88d30f4622a/libs/ubpf/kernel/ubpf_kernel.c#L31

uBPF prints messages via fprintf by default.

**Review by dthaler — APPROVED** [review]

**Review by shankarseal — APPROVED** [review]

File: tests/api_test/api_test.cpp [hunk 3]
```diff
@@ -288,8 +292,25 @@ ring_buffer_api_test(ebpf_execution_type_t execution_type)
...
 }
 
+// See also divide_by_zero_test_um in end_to_end.cpp for the user-mode equivalent.
+void
+divide_by_zero_test_km(ebpf_execution_type_t execution_type)
+{
+    struct bpf_object* object = nullptr;
+    hook_helper_t hook(EBPF_ATTACH_TYPE_BIND);
+    program_load_attach_helper_t _helper(
+        "divide_by_zero.o", EBPF_PROGRAM_TYPE_BIND, "divide_by_zero", execution_type, nullptr, 0, hook);
```
Comment (shankarseal) [inline tests/api_test/api_test.cpp:302]: maybe change the section string of `divide_by_zero.c` as well?
